### ISSUE TITLE
Add loadable_variables='coords' option

### ISSUE THREAD

**Issue by TomNicholas** [zarr-developers/VirtualiZarr#335]
> > is there a reason one would not want to always include all coordinates in the `loadable_variables` 
> 
> The main reason though is that coordinates can be N-dimensional in general, in which case you might use a lot of storage duplicating them. But for low-dimensional coordinates I agree we should recommend people load them.
> 
> Perhaps we should even add another option: `loadable_variables='coords'` or `loadable_variables='1d_coords'`, which we could default to... The latter is pretty much what `xr.open_dataset` already does for choosing which coordinates to create indexes for.

 _Originally posted by @TomNicholas in [#330](https://github.com/zarr-developers/VirtualiZarr/issues/330#issuecomment-2524060745)_

**Comment by TomNicholas** [comment]
In fact if we do this then why not just change the default for `indexes` too, or even get rid of it entirely. Loading 1D coordinates and creating indexes, but keeping 2+D coordinates virtual and not trying to create indexes for them (which I don't think is even possible in xarray right now anyway) would be much more intuitive and sane default behaviour, and it would mean all our examples don't need the `indexes={}` kwarg in them anymore. 

In other words we want the default behaviour to be the same as xarray except that data variables and multi-dimensional coordinate variables are virtual. I think I was just over-complicating things.

**Comment by TomNicholas** [comment]
In #353 I added the options, but making them work with the `indexes` kwarg for all possible cases is a lot of effort for basically no gain as far as I can tell. I think I will just remove `indexes` entirely.
